BigQuery/src/BigQueryClient.php

```diff
@@ -377,7 +390,7 @@ public function runQuery(JobConfigurationInterface $query, array $options = [])
      * @param array $options [optional] Configuration options.
      * @return Job
      */
-    public function startQuery(JobConfigurationInterface $query, array $options = [])
+    public function startQuery(QueryJobConfiguration $query, array $options = [])
```

Comment:
While there's an argument that the type was incorrect, narrowing it is a breaking change. Would any `JobConfigurationInterface` type work when given to this method? Unless other types cause errors, it should allow any interface type.

Reply:
You're right. `startQuery()` can accept any job configuration with no problems. But `runQuery()` throws a "not a query job" when wrong job configuration type is passed. I suggest widening argument type for `startQuery()` and narrowing it for `runQuery()` instead.

Reply:
Let's just leave it as-is (both should type-hint `JobConfigurationInterface`). Though you are correct that `runQuery` using `QueryJobConfiguration` doesn't work, the documentation is correct and changing it would change error handling semantics, especially in pre-PHP 7 environments. Since we haven't had any comments on it until now, I think it's best to leave well-enough alone, at least until there's a major version bump.